docs: improve self hosting analytics documentatoin (#16440)
supabase/supabase: apps/docs/docs/ref/self-hosting-analytics/introduction.mdx

All Logflare technical documentation is available at https://docs.logflare.app.

</Admonition>

## Backends Supported

The Analytics server supports either **Postgres** or **BigQuery** as the backend. The `supabase-cli` experience uses the Postgres backend out-of-the-box. However, the Supabase Platform uses the BigQuery backend for storing all platform logs.

When using the BigQuery backend, a BigQuery dataset is created in the provided Google Cloud project, and tables are created for each source. Log events are streamed into each table, and all queries generated by Studio or by the Logs Explorer are executed against the BigQuery API. This backend requires internet access to work, and cannot be run fully locally.

When using the Postgres backend, tables are created for each source within the provided schema (for supabase-cli, this would be `_analytics`). Log events received by Logflare are inserted directly into the respective tables. All BigQuery-dialect SQL queries from Studio will be handled by a translation layer within the Analytics server. This translation layer translates the query to PostgreSQL dialect, and then executes it against the Postgres database.

> The Postgres backend is not yet optimized for high volume of inserts, or for heavy query usage. Furthermore, the translation layer only handles a limited subset of the BigQuery dialect such that the Studio is capable to work as per normal. As such, the [Log Explorer](https://supabase.com/docs/guides/platform/logs#logs-explorer) may have errors for more advanced queries when using the Postgres Backend.

Reviewer: ```suggestion
The Postgres backend is not yet optimized for a high volume of inserts, or for heavy query usage. Today the translation layer only handles a limited subset of the BigQuery dialect. As such, the [Log Explorer](https://supabase.com/docs/guides/platform/logs#logs-explorer) may produce errors for more advanced queries when using the Postgres Backend.
```

## Getting Started

> The Postgres backend is recommended for getting familiar and experimenting with self-hosting Supabase. However, it is recommended to use the BigQuery backend for production scale. See [production recommendations](#production-recommendations) for more information

Reviewer: ```suggestion
The Postgres backend is recommended when familiarizing and experimenting with self-hosting Supabase. For production, we recommend using the BigQuery backend. See [production recommendations](#production-recommendations) for more information.
```

> To get started with self-hosting Supabase, the [example docker-compose](https://github.com/supabase/supabase/tree/master/docker) will give you a jumpstart to setting up logging for the stack.

Reviewer: ```suggestion
To set up logging in self-hosted Supabase, see the [docker-compose example](https://github.com/supabase/supabase/tree/master/docker).
```

Two compose services are required: Logflare, and Vector. Logflare is the HTTP Analytics server, while Vector is the logging pipeline to route all compose services' syslog to the Logflare sever.

> Regardless of backend chosen, the following environment variables **must** be set for the `supabase/logflare` docker image:

Reviewer: ```suggestion
Regardless of the backend chosen, the following environment variables **must** be set for the `supabase/logflare` docker image:
```

### Configuration and Requirements

The requirements are as follows after creating the project:

- Google Cloud project with billing enabled
- Project ID
- Project number
- A service account key.

<Admonition>

You must enable billing on your Google Cloud project, as a valid billing account is required for streaming inserts.

</Admonition>

#### Setting up BigQuery Service Account

The service account used must have sufficient permissions to insert into your Google Cloud BigQuery. Ensure that the service account has either:

- BigQuery Admin role; or
- The following permissions:
  - `bigquery.datasets.create`
  - `bigquery.datasets.get`
  - `bigquery.datasets.getIamPolicy`
  - `bigquery.datasets.update`
  - `bigquery.jobs.create`
  - `bigquery.routines.create`
  - `bigquery.routines.update`
  - `bigquery.tables.create`
  - `bigquery.tables.delete`
  - `bigquery.tables.get`
  - `bigquery.tables.getData`
  - `bigquery.tables.update`
  - `bigquery.tables.updateData`

> You can create the service account via the web console or `gcloud`, as per the [Google Cloud documentation](https://cloud.google.com/iam/docs/keys-create-delete). In the web console, you can create the key by navigating to IAM > Service Accounts > Actions (dropdown) > Manage Keys

Reviewer: ```suggestion
You can create the service account via the web console or `gcloud` CLI, as per the [Google Cloud documentation](https://cloud.google.com/iam/docs/keys-create-delete). In the web console, you can create the key by navigating to IAM > Service Accounts > Actions (dropdown) > Manage Keys
```

The current Postgres Ingestion backend isn't optimized for production usage. We recommend using Big Query for more heavy use cases.

> **We advise using the BigQuery backend for production settings, for better scaling and querying/debugging experience**.

Reviewer: ```suggestion
**We recommend using the BigQuery backend for production environments as it offers better scaling and querying/debugging experiences.**
```
